server/src/repositories/access.repository.ts

```diff
@@ -118,18 +119,21 @@ class AlbumAccess implements IAlbumAccess {
 
   @GenerateSql({ params: [DummyValue.UUID, DummyValue.UUID_SET] })
   @ChunkedSet({ paramIndex: 1 })
-  async checkSharedAlbumAccess(userId: string, albumIds: Set<string>): Promise<Set<string>> {
+  async checkSharedAlbumAccess(userId: string, albumIds: Set<string>, readWrite: ReadWrite): Promise<Set<string>> {
     if (albumIds.size === 0) {
       return new Set();
     }
 
     return this.albumRepository
       .find({
         select: { id: true },
+        relations: { sharedUsers: true },
         where: {
           id: In([...albumIds]),
           sharedUsers: {
-            id: userId,
+            user: { id: userId },
```

Comment:
`sql:generate` still gives the following message here:

```
AccessRepository.album.checkSharedAlbumAccess error: QueryFailedError: invalid input syntax for type uuid: "{"id":"00000000-0000-4000-a000-000000000000"}"
```

Same thing happens when calling the endpoint. Strange, because this is what TS is happy with.

`relations: { sharedUsers: { user: true } },` doesn't help either.

@jrasm91 can you please take a look, I don't know how to proceed.

Reply:
Figured it out! 🙂

Found the solution here:
https://github.com/typeorm/typeorm/pull/8616#:~:text=You%20still%20have%20an%20ability%20to%20use%20foreign%20keys%20as%20your%20primary%20keys